rust/core-lib/src/plugins/rpc_types.rs

```diff
@@ -127,10 +127,35 @@ pub struct Span {
 }
 
 #[derive(Serialize, Deserialize, Debug)]
+pub struct ScopeSpan {
+    pub start: usize,
+    pub end: usize,
+    pub scope_id: u32,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Style {
+    pub fg: u32,
+    #[serde(rename = "font")]
+    pub font_style: u8,
```

Comment:
This is compat with some previously defined type somewhere. We have a bunch of redundant Style defs in various places, which will be ripped out at the end of this work.